### PR TITLE
Output format decoupled and parametrized. Added JSON output.

### PR DESCRIPTION
**Pull request by jion**
In this commit, I created an output interface in order to decouple the format in which the output is printed from the internal structure.
I put the previous output format inside _plainOutput_ implementation and also added _jsonOutput_ implementation who can be printed in two flavors: pretty (easy to read) and normal (without tabs and carriage returns).
Also, I added an argument (--output or -o) in order to switch between output formats.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -122,8 +124,26 @@ func main() {
 
 	diffs := compare(configs)
 
-	for key, val := range diffs {
-		fmt.Printf("%35s: %40s : %40s\n", key, val[0], val[1])
+	formatter, err := getFormatter(opts.OutputFmt[0])
```
Comment (percona-csalguero) [inline main.go]: This must be before calling compare

File: output_interface.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+type plainOutput struct {
+}
+
+func (o *plainOutput) Format(diff map[string][]interface{}) (string, error) {
+	var buffer bytes.Buffer
+	for key, val := range diff {
+		buffer.WriteString(fmt.Sprintf("%35s: %40s : %40s\n", key, val[0], val[1]))
+	}
+
+	return string(buffer.String()), nil
```
Comment (percona-csalguero) [inline output_interface.go]: Isn't `buffer.String()` already a string?

File: main.go [hunk 3]
```diff
@@ -16,6 +17,7 @@ import (
 type options struct {
 	CNFs        []string
 	DSNs        dsnFlags
+	OutputFmt   []string
```
Comment (percona-csalguero) [inline main.go]: There is only one possible output format at a time so, this should be a plain string.

File: main.go (hunk 1 above)
Comment (jion) [reply main.go]: Why should this be called before? I suppose is to be sure that the formatter can be loaded before making any work (calculate diff). Is my assumption right?

Anyway, I put it before.

Comment (percona-csalguero) [reply main.go]: Yes, to avoid doing work before showing an error saying you cannot display the results

**Pull request merged**